pymodbus/client/mixin.py

```diff
@@ -551,11 +551,11 @@ def convert_to_registers(
         """Convert int/float/str to registers (16/32/64 bit).
 
         :param value: value to be converted:
-        :param data_type: data type to convert to
+        :param data_type: data type to be encoded as registers
         :returns: List of registers, can be used directly in e.g. write_registers()
         """
-        if data_type == cls.DATATYPE.STRING:
-            byte_list = value.encode()  # type: ignore[union-attr]
+        if data_type == cls.DATATYPE.STRING and isinstance(value, str):
```

Comment:
I like the idea, but how about moving it inline and do a raise (as I do with the register count).
Something like:
```
if data_type == cls.DATATYPE.STRING:
    if not isinstance(value, str):
        raise RuntimeError(“wrong combination of datatype and value”)
```

Sorry I am on my ipad, so no nice formatting.

Reply:
your suggestion works, but the side effects can be dangerous, since it goes into the int/float part.